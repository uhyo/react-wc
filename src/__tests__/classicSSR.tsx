/**
 * @jest-environment node
 */
import React, { Fragment } from "react";
import { renderToStaticMarkup, renderToString } from "react-dom/server";
import { ServerRenderingCollector, wc } from "..";

describe("Classic SSR", () => {
  describe("wc", () => {
    it("one slot", () => {
      const Hello = wc({
        shadowHtml: `
          <style>
            div {
              font-size: 100px;
            }
          </style>
          <div><slot></slot></div>
      `,
        slots: [],
        name: "wc-test-foo",
        classicSSR: true,
      });

      const collector = new ServerRenderingCollector();
      const str = renderToString(
        collector.wrap(
          <Hello>
            <span>Foobar</span>
          </Hello>
        )
      );

      expect(str).toMatchSnapshot();
      expect(
        renderToStaticMarkup(<Fragment>{collector.getHeadElements()}</Fragment>)
      ).toMatchSnapshot();
    });
    it("named slot", () => {
      const Hello = wc({
        shadowHtml: `
          <style>
            div {
              font-size: 100px;
            }
          </style>
          <slot name="foo"></slot>
      `,
        slots: ["foo"],
        name: "wc-test-2",
        classicSSR: true,
      });

      const collector = new ServerRenderingCollector();
      const str = renderToString(
        collector.wrap(<Hello foo={<span>Hi</span>} />)
      );

      expect(str).toMatchSnapshot();
      expect(
        renderToStaticMarkup(<Fragment>{collector.getHeadElements()}</Fragment>)
      ).toMatchSnapshot();
    });
    it("many components", () => {
      const Hello = wc({
        shadowHtml: `
          <style>
            div {
              font-size: 100px;
            }
          </style>
          <slot name="foo"></slot>
          <slot></slot>
      `,
        slots: ["foo"],
        name: "wc-test-many-components-1",
        classicSSR: true,
      });
      const Hello2 = wc({
        shadowHtml: `
          <style>
            :host { color: red; }
          </style>
          <slot name="foo"></slot>
      `,
        slots: ["foo"],
        name: "wc-test-many-components-2",
        classicSSR: true,
      });

      const collector = new ServerRenderingCollector();
      const str = renderToString(
        collector.wrap(
          <Hello foo={<span>Hi</span>}>
            <div>
              <Hello foo="foo" />
            </div>
            <Hello2 foo={<span>Hey</span>} />
          </Hello>
        )
      );

      expect(str).toMatchSnapshot();
      expect(
        renderToStaticMarkup(<Fragment>{collector.getHeadElements()}</Fragment>)
      ).toMatchSnapshot();
    });
  });
});
