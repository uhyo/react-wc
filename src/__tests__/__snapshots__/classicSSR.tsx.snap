// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Classic SSR wc many components 1`] = `"<wc-test-many-components-1><span slot=\\"foo\\">Hi</span><div><wc-test-many-components-1><wc-text slot=\\"foo\\">foo</wc-text></wc-test-many-components-1></div><wc-test-many-components-2><span slot=\\"foo\\">Hey</span></wc-test-many-components-2></wc-test-many-components-1>"`;

exports[`Classic SSR wc many components 2`] = `"<script>var __REACT_WC__={c:()=>class E extends HTMLElement{constructor(){super();this.attachShadow({mode:\\"open\\"}).appendChild(E.template.cloneNode(true))}refresh(){let c,s=this.shadowRoot;if(s){while(c=s.firstChild)s.removeChild(c);s.appendChild(E.template.cloneNode(true))}}},p:(t,d)=>{return d=document.createElement(\\"div\\"),d.insertAdjacentHTML(\\"afterbegin\\",t),t=document.createDocumentFragment(),t.appendChild(...d.childNodes),t}};customElements.define(\\"wc-test-many-components-1\\",(E=>(E.template=__REACT_WC__.p(\\"\\\\n          <style>\\\\n            div {\\\\n              font-size: 100px;\\\\n            }\\\\n          </style>\\\\n          <slot name=\\\\\\"foo\\\\\\"></slot>\\\\n          <slot></slot>\\\\n      \\"),E))(__REACT_WC__.c()));customElements.define(\\"wc-test-many-components-2\\",(E=>(E.template=__REACT_WC__.p(\\"\\\\n          <style>\\\\n            :host { color: red; }\\\\n          </style>\\\\n          <slot name=\\\\\\"foo\\\\\\"></slot>\\\\n      \\"),E))(__REACT_WC__.c()));</script>"`;

exports[`Classic SSR wc named slot 1`] = `"<wc-test-2><span slot=\\"foo\\">Hi</span></wc-test-2>"`;

exports[`Classic SSR wc named slot 2`] = `"<script>var __REACT_WC__={c:()=>class E extends HTMLElement{constructor(){super();this.attachShadow({mode:\\"open\\"}).appendChild(E.template.cloneNode(true))}refresh(){let c,s=this.shadowRoot;if(s){while(c=s.firstChild)s.removeChild(c);s.appendChild(E.template.cloneNode(true))}}},p:(t,d)=>{return d=document.createElement(\\"div\\"),d.insertAdjacentHTML(\\"afterbegin\\",t),t=document.createDocumentFragment(),t.appendChild(...d.childNodes),t}};customElements.define(\\"wc-test-2\\",(E=>(E.template=__REACT_WC__.p(\\"\\\\n          <style>\\\\n            div {\\\\n              font-size: 100px;\\\\n            }\\\\n          </style>\\\\n          <slot name=\\\\\\"foo\\\\\\"></slot>\\\\n      \\"),E))(__REACT_WC__.c()));</script>"`;

exports[`Classic SSR wc one slot 1`] = `"<wc-test-foo><span>Foobar</span></wc-test-foo>"`;

exports[`Classic SSR wc one slot 2`] = `"<script>var __REACT_WC__={c:()=>class E extends HTMLElement{constructor(){super();this.attachShadow({mode:\\"open\\"}).appendChild(E.template.cloneNode(true))}refresh(){let c,s=this.shadowRoot;if(s){while(c=s.firstChild)s.removeChild(c);s.appendChild(E.template.cloneNode(true))}}},p:(t,d)=>{return d=document.createElement(\\"div\\"),d.insertAdjacentHTML(\\"afterbegin\\",t),t=document.createDocumentFragment(),t.appendChild(...d.childNodes),t}};customElements.define(\\"wc-test-foo\\",(E=>(E.template=__REACT_WC__.p(\\"\\\\n          <style>\\\\n            div {\\\\n              font-size: 100px;\\\\n            }\\\\n          </style>\\\\n          <div><slot></slot></div>\\\\n      \\"),E))(__REACT_WC__.c()));</script>"`;
